src/csharp/Grpc.Core/GrpcEnvironment.cs

```diff
@@ -67,6 +62,21 @@ public class GrpcEnvironment
         readonly DebugStats debugStats = new DebugStats();
         bool isClosed;
 
+
+        /// <summary>
+        /// This is provided for advanced use-cases, such as libraries which provide a layer of abstract over grpc.
+        /// Setting this value will not have an effect until after all open channels/servers are shutdown.
+        /// Therefore, an application may change this at anytime but will need to orchastrate restrarting all channels/servers.
+        /// </summary>
+        /// <remarks>The default is 4</remarks>
+        /// <returns>The previous thread pool size</returns>
+        public static int SetThreadPoolSize(int numberOfThreads)
```

Comment:
I can't accept such change until we do more work on performance testing (as the internals of the thread pool used might change and then we would need to break the API).

Also, this is a logically separate change, so there should be a separated PR for it (but as I said, I can't probably accept this yet).